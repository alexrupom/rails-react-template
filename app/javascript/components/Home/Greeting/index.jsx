import React from 'react';
import PropTypes from 'prop-types';
import Alert from 'react-bootstrap/Alert';

const Greeting = ({ name }) => (
  <div className="col-grid">
    <div className="col-12">
      <Alert variant="primary" className="greeting__name">
        Hello {name}
      </Alert>
    </div>
  </div>
);

Greeting.defaultProps = {
  name: 'Alex',
};

Greeting.propTypes = {
  name: PropTypes.string,
};

export default Greeting;
