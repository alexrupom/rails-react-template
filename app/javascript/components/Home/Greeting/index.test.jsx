import React from 'react';
import { shallow } from 'enzyme';
import Greeting from '.';

describe('Greeting', () => {
  const props = {
    name: 'John Doe',
  };

  const output = shallow(<Greeting {...props} />);

  test('renders component', () => {
    expect(output).toMatchSnapshot();
  });
});
