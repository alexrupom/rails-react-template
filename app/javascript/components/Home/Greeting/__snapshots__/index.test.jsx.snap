// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Greeting renders component 1`] = `
<div
  className="col-grid"
>
  <div
    className="col-12"
  >
    <Alert
      className="greeting__name"
      closeLabel="Close alert"
      show={true}
      transition={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "appear": false,
            "in": false,
            "mountOnEnter": false,
            "timeout": 300,
            "unmountOnExit": false,
          },
          "displayName": "Fade",
          "render": [Function],
        }
      }
      variant="primary"
    >
      Hello 
      John Doe
    </Alert>
  </div>
</div>
`;
